pkg/secretlock/env/env_secret_lock.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package env
+
+import (
+	"crypto/cipher"
+	"encoding/base64"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/google/tink/go/subtle/random"
+	"golang.org/x/crypto/chacha20poly1305"
+
+	"github.com/hyperledger/aries-framework-go/pkg/secretlock"
+)
+
+// Lock is a secret lock service responsible for encrypting keys using a master key stored as environment variable
+type Lock struct {
+	aead cipher.AEAD
+}
+
+// NewService creates a new instance of a local secret lock service
+func NewService(keyURI string) (secretlock.Service, error) {
+	mk := os.Getenv("LOCAL_" + strings.ReplaceAll(keyURI, "/", "_"))
+	if mk == "" {
+		return nil, fmt.Errorf("masterKey not set")
+	}
+
+	masterKey, err := base64.URLEncoding.DecodeString(mk)
+	if err != nil {
+		return nil, err
+	}
+
+	aead, err := chacha20poly1305.NewX(masterKey)
```

Comment:
cannot hard-code to chacha.

why not AES GCM?

Reply:
this is internal to framework, the user should not offered to select the key wrapping algorithm for local storage, or do you think otherwise?

chacha is selected for faster/shorter key wrapping ops

Reply:
switched to AES GCM